### ISSUE TITLE
NPC on faction camp missions will starve even with food in their inventory.

### ISSUE THREAD

**Issue by Quintovir** [CleverRaven/Cataclysm-DDA#33661]
# Describe the bug
NPCs on missions that last a week or more can visibly be seen starving in the Factions (#) > Your Followers screen even though they have food in their inventory.
When I recall my NPC named Byron 'Amethyst' Hoyt from the Expansion Blacksmith mission I can see that he has plenty of food in his inventory and his weight is "Skeletal".
It's possible that NPCs on shorter build missions also do not eat, but they complete their missions before they starve themselves.
I've had 2 NPCs on Gathering missions for 1 week and 4 weeks and they have "Nominal" hunger values.

# Steps To Reproduce
1. Fill NPCs' inventories with non-perishable food (I give them smoked meat).
2. Send NPCs on expansion build mission that last a week or longer.
3. Watch their hunger progressively get worse.
4. When the NPCs have completed their build missions check their inventories and character pages.

# Expected behavior
NPCs should not starve with acceptable/sufficient food in their inventory. It might be simpler to remove an additional amount of kcal from the "Total faction food stock" on top of the kcal you pay them for going on a mission instead of requiring them to have food in their inventories.

# Screenshots
N/A

# Versions and configuration
- OS: Linux
    - OS Version: LSB Version: n/a; Distributor ID: ManjaroLinux; Description: Manjaro Linux; Release: 18.0.4; Codename: Illyria; 
- Game Version: 0.D-7064-g6831dd6e4f-dirty [64-bit]
- Graphics Version: Tiles
- Mods loaded: [
    Dark Days Ahead [dda],
    Makeshift Items Mod [makeshift],
    Medieval and Historic Classes and Shields [Medieval_Stuff],
    More Survival Tools [More_Survival_Tools],
    Modular Turrets [modular_turrets],
    Salvaged Robots [Salvaged_Robots],
    Hydroponics [hydroponics],
    Mutant NPCs [mutant_npcs],
    Beta National Guard Camp [national_guard_camp],
    More Locations [more_locations],
    Fuji's More Buildings [FujiStruct],
    Vehicle Additions Pack [blazemod],
    Tanks and Other Vehicles [Tanks],
    Folding Parts pack [deoxymod],
    No Fictional Guns [no_rivtech_guns],
    No Flaming Weapons [no_flaming_weapons],
    Classes and Scenarios Mod [more_classes_scenarios],
    Garden Pots [growable-pots],
    No Fungal Monsters [No_Fungi],
    Stats Through Kills [stats_through_kills]
]

# Additional context
Save game with the starving NPC ready to be recalled from the completed mission:
[West Jefferson.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/3557404/West.Jefferson.zip)

**Comment by mlangsdorf** [comment]
NPCs away on missions aren't actually starving, but it's a display bug that should get fixed

**Comment by Quintovir** [comment]
> NPCs away on missions aren't actually starving, but it's a display bug that should get fixed
You're right. I looked at an earlier backup where Byron 'Amethyst' Hoyt was at a nominal weight and wasn't malnourished and his stats are the same/unaffected as when he has a malnourished effect in his character sheet.

**Comment by mlangsdorf** [comment]
This is a good first issue for someone who knows how to program.

**Comment by y2s82** [comment]
Where would one start to address the wrong state display issue?

**Comment by korvyr** [comment]
Tagging this for my plate this coming weekend (or sooner). Seems simple enough at a glance.
